core/src/main/java/com/novoda/accessibility/ActionsAlertDialogCreator.java

```diff
@@ -8,28 +8,25 @@
 public class ActionsAlertDialogCreator {
 
     private static final int NO_TITLE = 0;
-    
+
     private final Context context;
 
     @StringRes
     private final int title;
 
-    private final Actions actions;
-
-    public ActionsAlertDialogCreator(Context context, Actions actions) {
-        this(context, NO_TITLE, actions);
+    public ActionsAlertDialogCreator(Context context) {
+        this(context, NO_TITLE);
     }
 
-    public ActionsAlertDialogCreator(Context context, @StringRes int title, Actions actions) {
+    public ActionsAlertDialogCreator(Context context, @StringRes int title) {
         this.context = context;
         this.title = title;
-        this.actions = actions;
     }
 
-    public AlertDialog create() {
+    public AlertDialog create(final Actions actions) {
         AlertDialog.Builder builder = new AlertDialog.Builder(context);
```

Comment:
I wrote a yuge paragraph on why I didn't think this was a good idea, then convinced myself, maybe it IS a good idea.

Then I tried it but I would avoid it. AlertDialog.Builder keeps state (duh, it's a builder) which it passes to the dialog on creation (regardless whether that state was set before or after the latest call to `build()` (of course)), and I think we want a clean dialog each time.

At this time, it wouldn't be a big deal because we don't provide options to provide customisations beyond the title, which is specified per `ActionsAlertDialogCreator`, but I can see we might in future if we introduce/need to introduce analytics.

It'll be for things like setting an OnDismissListener, for example. If we set the listener on every item we create it's fine (even though it's not thread safe, given that these dialogs are most likely (or need to be, not sure?)) created on the main/UI thread) since we'll replace the previous state. I'm more worried about cases where not all items would have a dismiss listener, but because we reuse the same builder, it'll be set on the item regardless and produce unexpected behaviour.

---

On the other hand, you could argue that this class should only share instances for like-cases, e.g. if you have a case where one list item would have a certain dialog property but another one does not, then use two instances of ActionsAlertDialogCreator.

In either case, I would weigh the pro-con before making the change:

- newing the builder in the `create()` method seems more in line with expectation than in a static factory method - at least, as a client of the lib, I'd expect there to be object allocation in there.
- shared state in the alert dialog builder means we have to make assumptions about the usage of this class and then relay these assumptions to the users of this library
- better testability (and if it changes, the tests should be in the same PR else :rage4:) but atm, I have high confidence it works as expected since it's so small and there is little behaviour. If there were no cons, I would go for it, just for future confidence that the class behaves as expected too.